File: Java/AutomationFramework/src/com/app/helpers/actions/MouseActions.java
package com.app.helpers.actions;

import com.app.helpers.logging.ILogger;
import com.app.helpers.logging.InfoLogger;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.interactions.Actions;

public class MouseActions {
    Actions actions;
    WebDriver driver;

    MouseActions(WebDriver driver) {
        this.driver = driver;
        actions = new Actions(driver);
    }

    public void ClickElement(WebElement element) {
        MoveToElement(element);
        element.click();
        ILogger.PrintLog(new InfoLogger().LogMessage("Clicked on the element '" + element + "'"));
    }

    public void MoveToElement(WebElement element) {
        actions.moveToElement(element)
                .perform();
        ILogger.PrintLog(new InfoLogger().LogMessage("Moved to the element '" + element + "'"));
    }

    public void MouseHover(WebElement element) {
        MoveToElement(element);
        ILogger.PrintLog(new InfoLogger().LogMessage("Mouse hovered on the element '" + element + "'"));
    }

    public void RightClickElement(WebElement element) {
        MoveToElement(element);
        actions.contextClick()
                .perform();
        ILogger.PrintLog(new InfoLogger().LogMessage("Right clicked on the element '" + element + "'"));
    }
}

File: DotNet/Framework/AutomationFramework/Helpers/Actions/MouseActions.cs
using OpenQA.Selenium;
using Helpers.Logging;

namespace Helpers.Actions
{
    public class MouseActions
    {
        OpenQA.Selenium.Interactions.Actions Action;
        IWebDriver driver;

        public MouseActions(IWebDriver driver)
        {
            this.driver = driver;
        }

        public void ClickElement(By selector)
        {
            MoveToElement(selector);
            driver.FindElement(selector).Click();
            Logger.PrintLog(new InfoLogger().LogMessage($"Clicked on the element '{selector}'"));
        }

        protected void MoveToElement(By selector)
        {
            Action = new OpenQA.Selenium.Interactions.Actions(driver);
            Action.MoveToElement(driver.FindElement(selector))
                .Perform();
            Logger.PrintLog(new InfoLogger().LogMessage($"Moved to the element '{selector}'"));
        }

        public void MouseHover(By selector)
        {
            MoveToElement(selector);
            Logger.PrintLog(new InfoLogger().LogMessage($"Mouse hovered on the element '{selector}'"));
        }

        public void RightClick(By selector)
        {
            MoveToElement(selector);
            Action = new OpenQA.Selenium.Interactions.Actions(driver);
            Action.ContextClick(driver.FindElement(selector))
                .Perform();
            Logger.PrintLog(new InfoLogger().LogMessage($"Right clicked on the element '{selector}'"));
        }
    }
}
